### ISSUE TITLE
Unhandled Exception when running a second ganeti-noded

### ISSUE THREAD

**Issue by Ganeti-Issues-Migrator** [ganeti/ganeti#256]
Originally reported of Google Code with ID 200.

```
What software version are you running? Please provide the output of "gnt-
cluster --version" and "gnt-cluster version".

Software version: 2.4.1
Internode protocol: 2040000
Configuration format: 2040000
OS api version: 20
Export interface: 0

gnt-cluster (ganeti v2.4.1) 2.4.1


What distribution are you using? RHEL6.1

<b>What steps will reproduce the problem?</b>
1. Start ganeti-noded as part of a healthy cluster
2. Execute /usr/sbin/ganeti-noded -f as the root user.

<b>What is the expected output? What do you see instead?</b>

I would expect some code to catch this exception and fail gracefully.

Instead I see:

[root@kvm1.build.mtv1 ~]# ganeti-noded -f
Traceback (most recent call last):
  File "/usr/sbin/ganeti-noded", line 19, in <module>
    sys.exit(main.Main())
  File "/usr/lib/python2.6/site-packages/ganeti/server/noded.py", line 1031, in Main
    console_logging=True)
  File "/usr/lib/python2.6/site-packages/ganeti/daemon.py", line 682, in GenericMain
    utils.WritePidFile(utils.DaemonPidFileName(daemon_name))
  File "/usr/lib/python2.6/site-packages/ganeti/utils/io.py", line 703, in WritePidFile
    filelock.LockFile(fd_pidfile)
  File "/usr/lib/python2.6/site-packages/ganeti/utils/filelock.py", line 45, in LockFile
    raise errors.LockError("File already locked")
ganeti.errors.LockError: File already locked


<b>Please provide any additional information below.</b>

```
Originally added on 2011-10-07 22:29:17 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
Hi,

Well, the code does fail "gracefully", as in refusing to run. I guess you wanted something more along the lines of:
# ganeti-noded -f
Error: Lockfile already locked, a second node daemon is already running.

Or similar?
```
Originally added on 2011-10-10 08:24:33 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
The condition I'm seeing is from abrtd's perspective it looks like a nasty error (an unhandled exception):

Oct  5 03:30:01 relabs05 python: abrt: detected unhandled Python exception in /usr/sbin/ganeti-noded
Oct  5 03:30:01 relabs05 abrtd: dumpsocket: New client connected
Oct  5 03:30:01 relabs05 abrtd: dumpsocket: Saved Python crash dump of pid 25238 to /var/spool/abrt/pyhook-1317810601-25238
Oct  5 03:30:01 relabs05 abrtd: dumpsocket: Socket client disconnected
Oct  5 03:30:01 relabs05 abrtd: Directory 'pyhook-1317810601-25238' creation detected
Oct  5 03:30:01 relabs05 abrtd: Crash is in database already (dup of /var/spool/abrt/pyhook-1313566201-10803)
Oct  5 03:30:01 relabs05 abrtd: Deleting crash pyhook-1317810601-25238 (dup of pyhook-1313566201-10803), sending dbus signal
```
Originally added on 2011-10-14 08:23:42 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
I hope the patch mentioned below will fix this for your usecase. It will be included in Ganeti 2.5.1 or 2.6. The new output looks like this:

# ganeti-noded -d -f
Error while locking PID file:
PID file '/var/run/ganeti/ganeti-noded.pid' is already locked by another process, PID read from file is 12577


commit 4958b41e8a4807c9c4dba519951a97aad53b4745
Author: Michael Hanselmann <hansmi@google.com>
Date:   Thu Dec 8 15:35:23 2011 +0100

    daemon.GenericMain: Don't generate backtrace on conflicting daemons
    
    Instead, print a nicer error message. This should fix issue 200.
    
    Signed-off-by: Michael Hanselmann <hansmi@google.com>
    Reviewed-by: Iustin Pop <iustin@google.com>

```
Originally added on 2011-12-08 16:28:06 +0000 UTC.

Mentioned Issue with GoogleCode ID 200 has been migrated to GitHub with ID #256.
Changed State: Fixed

**Comment by Ganeti-Issues-Migrator** [comment]
```
Migrate old issues from 'fixed' to 'released'.
```
Originally added on 2012-10-22 09:16:17 +0000 UTC.

Changed State: Released
